masto-module/src/main/kotlin/au/rakka/java/masto-api/MastoNotifier.kt

```diff
@@ -0,0 +1,83 @@
+package au.rakka.java.mastoapi
+
+import au.com.skater901.wc3.api.core.service.GameNotifier
+
+import java.net.http.HttpClient // This should be significantly less hassle.
+import java.net.http.HttpRequest
+import java.net.http.HttpResponse
+import java.net.URI
+
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentMap
+import au.com.skater901.wc3.api.core.domain.Game
+import au.com.skater901.wc3.api.core.domain.GameSource
+import au.com.skater901.wc3.api.core.domain.Region
+
+// Used to get my config class in here
+import jakarta.inject.Inject
+
+import kotlinx.coroutines.future.await
+import com.fasterxml.jackson.core.JsonParser
+import com.fasterxml.jackson.databind.ObjectMapper
+
+import org.slf4j.LoggerFactory
+
+public class MastoNotifier @Inject constructor(private val conf:MastoConfig) : GameNotifier
+{
+	private val hostedGameMessages: ConcurrentMap<Int, String> = ConcurrentHashMap()
+	private val hostedGameTags: ConcurrentMap<Int, String> = ConcurrentHashMap()
+	
+	private val logger = LoggerFactory.getLogger(MastoNotifier::class.java)
+	private val mapper = ObjectMapper()
+	
+	private val userAgent = "WC3 Notification Bot ${System.getProperty("appVersion")} - Java-http-client/${System.getProperty("java.version")}"
+	private val client: HttpClient = HttpClient.newHttpClient()
+	private val builder: HttpRequest.Builder = HttpRequest.newBuilder().setHeader("Authorization","Bearer ${conf.token}").setHeader("Content-Type","application/json").setHeader("User-Agent",userAgent)
+	
+	private val baseurl = "https://${conf.instance}/api/v1/" // Might want this for other endpoints later maybe
+	private val statusurl = "${baseurl}statuses"
+	
+	override suspend fun notifyNewGame(notificationId: String, game: Game)
+	{
+		logger.debug("Blep")
+		hostedGameTags[game.id]=notificationId
+		val response = client.sendAsync(builder.uri(URI.create(statusurl)).POST(game_to_string(game,hostedGameTags[game.id])).build(),HttpResponse.BodyHandlers.ofInputStream()).await()
+		logger.debug(response.statusCode().toString())
+		val node=mapper.readTree(response.body())
+		// Can't read it twice and can't be fucked figuring out how to stash it
+		//logger.debug(response.body().use { r -> r.use { r.reader().use { i -> i.readText() } } })
+		hostedGameMessages[game.id]=node.get("id").asText()
+		logger.debug("Saved ID " + game.id + ": "+hostedGameMessages[game.id])
```

Comment:
For loggers, it's better to use the logger's method of string interpolation. If the log level is set higher than debug, the logger won't do the string interpolation, so it saves CPU. It's pretty easy to do; use a pair of curly braces where you want the value to be, then provide the values in order after the log string. So this code would look like this: `logger.debug("Saved ID {}: {}", game.id, hostedGameMessages[game.id])`